netlify-templates/one-click-hugo-cms
# fix: Update the targetPort and update baseURL

## Summary

We had two issues come up with the final changes for dependencies which only came up for `netlify dev`:
1. If a user tries to boot up the project with `netlify-cli@10.6` and above we noticed the default expected port wasn't `3000`
2. If a user modifies the `targetPort` to be correct, the baseURL would be set incorrectly locally but not on the production contexts 

## Technical Details

If we wanted to solve the first issue we realized we need to modify the `[dev]` context a bit. After explicitly adding the `[dev]` key with a target framework we saw this might resolve the issue.

We also tried to explicitly add a `targetPort`, this was because [in the documentation it states that the CLI will wait till the targetPort is reachable](https://docs.netlify.com/configure-builds/file-based-configuration/#netlify-dev). We received this error which seems to validate the idea of using framework instead:
```
Add 'framework = "hugo"' to the [dev] section of your netlify.toml to avoid this selection prompt next time
```

After we did that, we saw an issue where the project wouldn't render the images whatsoever and that was due to the `<base>` tag in our `site/layouts/partials/head.html` wasn't able to get the appropriate `href` locally.

### Test Plan

- Clone the project
- Install dependencies
- Run `ntl dev`
- Verify that content is styled and images populate
